cql3/selection/selection.cc

```diff
@@ -644,6 +644,21 @@ bool result_set_builder::restrictions_filter::do_filter(const selection& selecti
 
     auto static_and_regular_columns = expr::get_non_pk_values(selection, static_row, row);
 
+    if (_partition_key_filter) {
+        if (!expr::is_satisfied_by(
+                        *_partition_key_filter,
+                        expr::evaluation_inputs{
+                            .partition_key = partition_key,
+                            .clustering_key = clustering_key,
```

Comment:
Why does the partition-key filter need the real clustering key? Or it's just easier (?) to pass the real one instead of some fake empty one?

Reply:
It does not need it, but it's a goal to remove the separation between partition keys and clustering keys during filtering and just have filters (more than one since some filters can be applied less often than others).